### DISCUSSION TITLE
Setting password for new users on creation

### DISCUSSION THREAD

**Discussion by catholic-indulgence-vaper** [lldap/lldap#749] [Q&A] [answered]
Hi!
Is there any way to set password using API on user creation or right after?
I can't see password setting examples in UI during API calls except `/auth/opaque/register/start` call.
Any ideas how can I do that?

**Comment by nitnelave** [comment]
Hi!
There's no way to set the password with graphql, since we use OPAQUE (a zero-knowledge protocol) to set the password hash without ever sending it to the server.
However, we provide a tool "lldap_set_password" to do just that.

**Comment by catholic-indulgence-vaper** [comment]
@nitnelave, so it's possible to form hash & to send password hash? :)
I need API call to do that.

**Reply by nitnelave** [reply, marked as answer]
It's a bit more complicated than that, but it's essentially what the lldap_set_password binary is doing. It's using OPAQUE to "hash the password and send the hash". So rather than sending an http request, you'd call the binary and it will send the requests for you.

**Reply by catholic-indulgence-vaper** [reply]
Thank you!
